### DISCUSSION TITLE
Intriguing estimation of substitution rates in IQTREE

### DISCUSSION THREAD

**Discussion by rbjleroy** [iqtree/iqtree2#209] [Q&A] [answered]
Hi,

I have been working on a phylogenomic analysis for quite some time now, and I conducted a progressive removal of the fastest-evolving sites in my alignment.

For this, I inferred site-specific substitution rates using IQTREE v2.0.3 (flag --rate, substitution model LG+C60+G). However, when looking at the profile of the rates inferred, they appeared to sort of follow the Gamma categories (see "Rates from IQTREE"). To be more precise, the rate value of the plateaux exactly match the gamma categories inferred during the tree search, but other sites have values in-between gamma category values.

I have tried using the ML-based rate inference method (flag --mlrate), where the distribution of the values is more as expected following an exponential-ish curve (see "ML Rates from IQTREE") although there is a plateau that I do find weird. But I do find that those rates are not comparable across inferences. Indeed, I inferred such ML rates on the same sites for slow- and fast-evolving taxa in my species tree separately, and compared the rates. With this, I find that slow-evolving taxa have more sites that evolve faster than in fast-evolving taxa than the opposite (see "ML Rates from IQTREE comparison").

When comparing these inferred rates with the ones inferred using Dist_Est (https://www.mathstat.dal.ca/~tsusko/doc/dist_est.pdf, see Rates from Dist_Est), the latter looked more plausible and comparable (see "Rates from Dist_est" and "Rates from Dist_Est comparison").

All of this to ask if there was something wrong in my command when inferring rates using IQTREE (I am maybe thinking of removing Gamma in the substitution model), and if you have already seen such a distribution of rates in the past?

Sorry for the humongous message and for the possible confusion. Don't hesitate to ask for anything (explanation or raw data), I remain available, and thank you very much for your time and your help! :)

Romain
![rates-problem](https://github.com/iqtree/iqtree2/assets/106535206/fdc231d0-30b7-415e-8278-c96a702f0e31)

**Comment by bqminh** [comment, marked as answer]
That's interesting observation indeed! I'm actually not aware of Dist_Est until now. Thanks for that. I can only comment on the perspective of two options you are using in IQ-TREE, noting that I don't really have any expectations as we don't know the ground truth.

`--rate` option (or `-wsr` in IQ-TREE 1) will apply the "empirical Bayesian approach" to compute the site rates using posterior probabilities of sites belonging to each of the rate categories (4 in this case as you are using `+G` which is a shorthand for `+G4`). From the output .rate file you can either grab the posterior mean or posterior mode. You are plotting the posterior mean here (top-left figure). This is _not_ unexpected, as there are 4 rate categories: there are 4 "horizontal lines" representing sites that have a very high probability of falling into one category, and some "slopes" in between for sites that can be in two (or even 3) categories with no "deciding" probability.

`--mlrate` option will not do this, but rather for each site, it finds a "scaling factor" of the tree length that would maximise the likelihood. That means, it tries to multiple every branch length of the ML tree with a scaling factor, and find the one with the highest likelihood. Sometimes, some sites may hit the upper bound of 100, meaning that IQ-TREE can't determine the rate of these sites because they are way too high (approaching +inf). That's what you see in the top-middle figure.

I have no knowledge of Dist_Est and thus can't comment on the top-right figure.

I don't know what are fast and slow-evolving taxa, thus also can't comment on other figures.

In summary, I don't see anything unexpected [about the options of IQ-TREE].

**Reply by rbjleroy** [reply]
Thank you very much for this response, that is really useful!! 

Best,
Romain
